diff --color -ru a/lua/presence/file_assets.lua b/lua/presence/file_assets.lua
--- a/lua/presence/file_assets.lua	2021-06-09 07:56:33.416958354 +0530
+++ b/lua/presence/file_assets.lua	2021-06-09 07:57:23.336181080 +0530
@@ -80,7 +80,7 @@
     hack = { "Hack", "hack" },
     hpp = { "C++ header file", "c_plus_plus" },
     hs = { "Haskell", "haskell" },
-    html = { "HTML", "html" },
+    html = { "HTML", "html", "htm" },
     hx = { "Haxe", "haxe" },
     hxx = { "C++ header file", "c_plus_plus" },
     ini = { "Configuration file", "config" },
diff --color -ru a/lua/presence/init.lua b/lua/presence/init.lua
--- a/lua/presence/init.lua	2021-06-10 09:03:09.696734425 +0530
+++ b/lua/presence/init.lua	2021-06-10 12:12:21.061551934 +0530
@@ -293,10 +293,10 @@
     end
 
     local env_vars = {
+        "XDG_RUNTIME_DIR",
         "TEMP",
         "TMP",
         "TMPDIR",
-        "XDG_RUNTIME_DIR",
     }
 
     for i = 1, #env_vars do
